### PR TITLE
Fix private link not accessible when disabling share API

### LINKED ISSUE #2598: [BUG] Private link not accesible
### Steps to reproduce
1. Disable Sharing API in web UI (only admins)
2. In the file list try to get the private link

### Actual behaviour
Not posible, since private link is inside the Sharing, that is disabled by capability.

### Expected behaviour
Link accesible
 
Options to fix:

1) Set the private link in the three-dot-button, independent from "Sharing"

Screenshot 2019-07-03 at 18 12 16

2) Option "Sharing" always visible. In case of API is disabled, only private link is displayed. This option leads to an almost blank screen:

Screenshot 2019-07-03 at 18 17 01

Which one do you like the most? more ideas welcome

@hosy @davigonz @michaelstingl 

### PR: 
- App: https://github.com/owncloud/android/pull/2617

### PR DESCRIPTION
**Pull request by davigonz**
Fixes #2598 

QA:

https://github.com/owncloud/android/pull/2617#issuecomment-518213737

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt [hunk 1]
```diff
@@ -176,6 +176,10 @@ class ShareFileFragment : Fragment(), ShareUserListAdapter.ShareUserAdapterListe
             )
         }
 
+    private val isShareApiEnabled: Boolean
+        get() = capabilities != null && capabilities?.filesSharingApiEnabled == CapabilityBooleanType.TRUE.value ||
```
Comment (jcmontesmartos) [inline owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt]: you can remove `capabilities != null` ;-)

File: owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt [hunk 2]
```diff
@@ -322,6 +326,17 @@ class ShareFileFragment : Fragment(), ShareUserListAdapter.ShareUserAdapterListe
         updatePublicLinkButton()
 
         // Update view depending on updated capabilities
+        if (!isShareApiEnabled) {
+            shareHeaderDivider.visibility = View.GONE
```
Comment (jcmontesmartos) [inline owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt]: why don't you use isVisible extension? This block would be refactored in:

        shareHeaderDivider.isVisible = isShareApiEnabled
        shareWithUsersSection.isVisible = isShareApiEnabled
        shareViaLinkSection.isVisible = isShareApiEnabled

File: owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt (hunk 1 above)
Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt]: Done

File: owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt (hunk 2 above)
Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/shares/presentation/fragment/ShareFileFragment.kt]: Done

**Comment by davigonz** [comment]
@jesmrec code approved, ready to test

**Comment by jesmrec** [comment]
QA checklist:

- [X] Share API enabled -> private link accesible
- [X] Share API disabled -> private link accesible

**Pull request merged**